models/cacao/variables.go

```diff
@@ -5,39 +5,94 @@ import (
 	"strings"
 )
 
-type Variable struct {
-	Type        string `bson:"type" json:"type" validate:"required"`
-	Name        string `bson:"name,omitempty" json:"name,omitempty"`
-	Description string `bson:"description,omitempty" json:"description,omitempty"`
-	Value       string `bson:"value,omitempty" json:"value,omitempty"`
-	Constant    bool   `bson:"constant,omitempty" json:"constant,omitempty"`
-	External    bool   `bson:"external,omitempty" json:"external,omitempty"`
+// Initialize new Variables
```

Comment:
@MaartendeKruijf is this closer to the interface you'd like to see? It's inspired by the CPP map interface.

Reply:
Looks good updated it with parsing json and pass by pointer